### DISCUSSION TITLE
Error from Version 1.2.0.0-RC4

### DISCUSSION THREAD

**Discussion by exercise42** [Royal-Modding/FS19_BetterContracts#35] [Q&A]
Hello Royal-Modding-Team,
big thanks for your great work. Stumbled into your wip-versions here, especially 1.2.0.0-RC4, whilst searching for a possibility to use mod-vehicles in missionvehicles.xml.
Is the mod still supported, even it's "old stuff" from FS19? Ran into 2 problems on the Mountain Hill 2021:
1. Creating a new savegame is doing well, but when I try to continue the save game, I'm told, there are no contracts available (even those, still working at at last save are missing). Maybe "fixed" it by adding the mod FS19_claasDominator108SL.zip (which is in your userdefined.xml by default). After adding this mod and picking up the savegame, the missions appear.
2. After some time of playing, the mod starts "spamming" the log contiuously with the following error:
"Error: I:/Games Installed/FarmingSimulator19mods/mods_MountainHill/FS19_BetterContracts/scripts/userint.lua:119: bad argument #1 to 'max' (number expected, got no value)"
(Edit: don't know, why after saving the errormessage is extended with a link, I haven't written it)
I can't tell you, if there is a special action that causes the error nor if there's a conflict with a script of one of the other mods....

I would be glad if you would still support this great mod on FS19. Do you have a Discord-channel?
Regards
exercise42

**Comment by Mmtrx** [comment]
Hi,
I'm afraid Royal Modding is no longer active in FarmSim. I cooperated with them on BetterContracts, and moved the mod to FS22 [FS22_BetterContracts](https://github.com/Mmtrx/FS22_BetterContracts).

Moving your gameplay over to FS22 is not an option I guess?
I can try to help you get around the current errors with the FS19 mod version. Speaking of which, what version are you really using? Did you copy the latest source files from here, and put them in a zip file?
Also, could you upload a copy of your log ?

**Reply by exercise42** [reply]
Hi Mmtrx,
thanks for joining! I appreciate your work (SeeBales as well as the extension to the mission-screen in the RC's of BC). No need to be afraid, to me it appears you are not less qualified to support here ;-))
And sadly, no, at the moment it's no option moving to FS22, as long as FS19 still provides a lot of fun ;-))
Might we continue in German language?
I'm using the last RC-Version created by you (1.2.0.0-RC4) (so I didn't download the sourcefiles and put them in zip).
When I load the savegame, there's no error at the start, but the mission-screen in the esc-menu is empty (see attachement).
Oh and a addition/correction to point 1. of my initial thread: Adding the claas-Dominator hasn't fixed the problem with the empty mission window.
Reverting back to version 1.1.0.0 shows missions in current savegame correctly, switching back to 1.2.0.0-RC4 still shows no missions.
Maybe it's a incompatibility with another (script-)mod?
Attached the corrupt mission-screen and a log
![BC](https://user-images.githubusercontent.com/86823712/230041961-37b9f340-537d-40f3-ae64-a4fa36bff962.jpg)
[log.txt](https://github.com/Royal-Modding/FS19_BetterContracts/files/11157552/log.txt)
Thanks in advance for your support!

**Reply by Mmtrx** [reply]
Moin,
ja wir können gern auf deutsch weitermachen.
Das Problem scheint an einem Missionsfahrzeug zu liegen, bei dem BC nicht die workwidth ermitteln kann. Kannst du auch noch mal die missions.xml aus dem savegame Ordner hochladen?

Eine schnelle Umgehungslösung wäre auch, diese missions.xml einfach zu löschen. Müsste man dann halt frische Missionen erzeugen.

Ich schreib kurz einen fix für dieses spezielle Problem, müsstest du aber selbst testen (hab kein FS19 mehr :).
[FS19_BetterContracts.zip](https://github.com/Royal-Modding/FS19_BetterContracts/files/11180763/FS19_BetterContracts.zip)

**Reply by exercise42** [reply]
Hallo Mmtrx,
das ist aber seltsam, denn ich wär' mir jetzt nicht bewusst ein Mod-Fahrzeug in irgendeiner missionVehicles verändert zu haben.... (also weder in der Map noch hier in BC)
Ich hab schon mal alle nicht begonnen Missionen in der missions.xml gelöscht gehabt, in der Hoffnung, daß es an einer der angebotenen Missionen bzw. der zugeordneten Fahrzeuge hakt, das hat aber nix gebracht.
Bei den angenommenen Missionen habe ich bei einigen Erntemissionen Fahrzeuge mit angenommen, die Düngemissionen fahre ich komplett mit eigenem Equipment.
Aber hier mal die Missions.xml als txt, da GitHub ja keine xml beim Hochladen unterstützt. Deine Version werd' ich mal testen, danke schon mal dafür und noch frohe Ostern!
Edit: Deine Anpassung funktioniert, die Anzeige der Missionen ist wieder da!!!!! Allerdings haut er jetzt 'ne andere Meldung raus:
2023-04-07 20:55   Warning: [FS19_BetterContracts]:getFromVehicle() could not find appropriate vehicle in mission 581. Last vehicle: SUGARCANEHARVESTING randon/canavieiro/canavieiro.xml
Und noch 'ne Frage (bevor ich lange teste, Du kennst sicher die Antwort): 
in der userVehicles.xml nutzt Du im vehicle-tag die Syntax
`<vehicle filename="vehicles/claas/dominator108SL/dominator108SL.xml" requiredMod="FS19_claasDominator108SL"> <!-- combine -->`
Ist die Kombination aus filename="..." und requiredMod="..." nur in der userVehicles.xml nutzbar oder auch in jeder missionVehicles.xml (wird das nur in BC korrekt interpretiert oder global)?
[missions.txt](https://github.com/Royal-Modding/FS19_BetterContracts/files/11180784/missions.txt)

**Comment by Mmtrx** [comment]
Ja, „requiredMod“ ist eine Erfindung von BC, wird also nicht in anderen missionVehicles.xml funktionieren. 

Die Warnung kann man ignorieren, er findet kein vehicle mit einer definierten workwidth, und kann daher die benötigte Weglänge für das Feld (und damit die Arbeitszeit) nicht abschätzen.

**Reply by exercise42** [reply]
Die Warning schreibt er aber nicht nur einmal raus, die kommt in einer Minute öfter. Mal sehen, wie schlimm das in der log wird.....
Auf jeden Fall herzlichen Dank für die Fehlerbereinigung.
Funktioniert die "requieredMod" innerhalb BC nur in der userVehicles oder auch in der missionVehicles? (ist echt schade, daß das nicht besser dokumentiert wurde, da steckt jetzt 'ne Menge Potential drin - endlich passende Geräte zur Mission....)

**Reply by Mmtrx** [reply]
Jo, die Nachricht kommt bei jedem internen update der Auftragsliste. Hab die Änderung zur Wiederholung der Warnung aus der BC 22er Version hier reinkopiert. Außerdem auch die "Anleitung" für userDefined.xml:
```
	BetterContracts user-defined mission vehicles
	Purpose:     Define vehicles to be used for contracts
	Author:      RoyalModding, Mmtrx   
	Usage:       You can define additional vehicle sets to use for specific contracts (aka missions).
			 You can define mod vehicles/ tools, see requiredMod="FS19_claasDominator108SL" below.
			 Your user-defined vehicles are added to the mission vehicles from base game (and possibly
			 from the map definition), to choose from when a new mission is created.

			 You can delete any base game or map-defined mission vehicles (and thus use only those
			 that you defined) by setting overwrite="true" in the <missionVehicles> tag. If you want to try 
			 this, use extreme
			 CAUTION: You must be sure that you specify a vehicle group for every mission type / fieldsize / 
					  variant combination possible. For mods, specify the correct modname and filename. 
					  BetterContracts will try to check this, and issue warnings in the log 
	Changelog:
	 v0.1        17.03.2021  initial by Royal-Modding
	 v1.1.0.4    07.07.2021  (Mmtrx) add user-defined missionVehicles.xml, allow missions with no vehicles
```
Du hast also volle Kontrolle über die Missionsfahrzeuge.
[FS19_BetterContracts.zip](https://github.com/Royal-Modding/FS19_BetterContracts/files/11183278/FS19_BetterContracts.zip)

**Reply by exercise42** [reply]
Hallo Mmtrx,
danke für die Ergänzungen. Zwar ist mir noch nicht klar, ob "filename=" und "requiredMod" auch in der in BC enthaltenen missionVehicle funktionieren, aber ich werd das mal testen. Die Option für mapspezifische missionVehicle.xml die Du/ihr in der RC4 geschaffen habt, würde die userDefined.xml ja obsolet machen, da ich nun für jede Map ein Set machen kann und ansonsten die in BC enthaltene missionVehicle zieht. Daß ich da noch jeweils die lua anpassen muss, ist mir bewusst, aber ich trau' es mir dann doch zu, den notwendigen Code (Abfrage der jeweiligen Map) anhand des bereits vorhandenen Beispiels (Thüringer Höhe) zu ergänzen.

Schönes Osterrestwochenende und nochmals vielen Dank!
